src/zc/buildout/download.py

```diff
@@ -31,17 +31,28 @@ class PatchedURLopener(FancyURLopener):
 
 except ImportError:
     # Python 2
+    import base64
     from urlparse import urlparse
-
+    from urlparse import urlunparse
     import urllib2
 
-    def urlretrieve(url, path):
-        """Work around Python issue 24599
+    def urlretrieve(url, tmp):
+        """Work around Python issue 24599 includig basic auth support
         """
-        url_obj = urllib2.urlopen(url)
-        with open(path, 'wb') as fp:
+        scheme, netloc, path, params, query, frag = urlparse(url)
+        auth, host = urllib2.splituser(netloc)
+        if auth:
+            url = urlunparse((scheme, host, path, params, query, frag))
```

Comment:
The `url = urlunparse(...)` and `req = urllib2.Request(url)` can be moved outside the "if" so that they're only needed once, right?

The "if auth" would then only need to compose and add the header. Little bit shorter that way.